### PR TITLE
[cinder-csi-plugin] Enable optional storageclasses.

### PR DESCRIPTION
**Pull request by powellchristoph**
[cinder-csi-plugin] Enable Optional StorageClasses

**What this PR does / why we need it**:
If you enable the included storageclasses, you get both the `retain` and `delete`. This PR allows each class to be enabled individually.

**Which issue this PR fixes(if applicable)**:
na

**Special notes for reviewers**:
na

**Release note**:
```release-note
[cinder-csi-plugin] Included storageclasses can be individually enabled.
```

### CODE REVIEW

**Comment by dulek** [comment]
@powellchristoph: Before jumping into the code - you need to retain the release note box from the PR template: https://github.com/kubernetes/cloud-provider-openstack/blob/66624e34b1884781dca75f2ae6dc9736c0de9f28/.github/PULL_REQUEST_TEMPLATE.md

File: charts/cinder-csi-plugin/values.yaml [hunk 1]
```diff
@@ -160,11 +160,12 @@ secret:
 #      ca-file=/etc/cacert/ca-bundle.crt
 
 storageClass:
-  enabled: true
```
Comment (dulek) [inline charts/cinder-csi-plugin/values.yaml:163]: Okay, this needs a release note for sure to show that the options are moved. I think I'd also keep the global option here and treat individual options as ones overriding the global one.

**Review by dulek — CHANGES_REQUESTED** [review]

**Comment by dulek** [comment]
I'm looking into chart lint failure.

**Comment by dulek** [comment]
See #2582 for chart linting failure explanation. It's not fault of this patch, we'll fix it soon.

/ok-to-test

**Comment by dulek** [comment]
Ah, looks like you need to make EasyCLA happy. Just click "Details" in its test table row and follow the steps.

File: charts/cinder-csi-plugin/values.yaml (hunk 1 above)
Comment (powellchristoph) [reply charts/cinder-csi-plugin/values.yaml:163]: I have updated the PR with the release notes and kept the default with an override.

**Comment by powellchristoph** [comment]
> Ah, looks like you need to make EasyCLA happy. Just click "Details" in its test table row and follow the steps.

I've followed the link and filled it out twice now. Its sent me the email. I'm not sure what else I'm supposed to do.

**Comment by dulek** [comment]
/approve

Can you squash the commits before the final `/lgtm`?

**Pull request closed without merging**